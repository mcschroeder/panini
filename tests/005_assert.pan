assert : { b:𝔹 | b = true } → 𝟙

f0 = assert true

f1 = assert false    -- expected error

f2 : unit
f2 = assert true

f3 : unit            -- expected error
f3 = assert false

f4 : {_:unit|?}
f4 = assert true

f5 : {_:unit|?}
f5 = assert false

f6 = \b:bool. assert b     -- expected error

f7 : bool -> unit          -- expected error
f7 = \b:bool. assert b

f8 : {b:bool|?} -> unit
f8 = \b:bool. assert b

f9 : {b:bool|?} -> unit
f9 = \b:bool. assert true

f10 : {b:bool|?} -> unit
f10 = \b:bool. assert false

f11 = let b = true in assert b

f12 = let b = false in assert b        -- expected error

f13 : unit
f13 = let b = true in assert b

f14 : unit                           -- expected error
f14 = let b = false in assert b

f15 : {_:unit|?}
f15 = let b = true in assert b

f16 : {_:unit|?}
f16 = let b = false in assert b

f17 = \s:string. assert false   -- expected error

f18 : {s:string|?} -> unit
f18 = \s:string. assert false

f19 : {s:string|false} -> unit
f19 = \s:string. assert false

f20 : {s:string|true} -> unit   -- expected error
f20 = \s:string. assert false

f21 : {s:string|s = "hello"} -> unit    -- expected error
f21 = \s:string. assert false

f22 = \s:string. assert true

f23 : {s:string|?} -> unit
f23 = \s:string. assert true

f24 : {s:string|true} -> unit
f24 = \s:string. assert true

f25 : {s:string|false} -> unit
f25 = \s:string. assert true

f26 : {s:string|s = "hello"} -> unit
f26 = \s:string. assert true
