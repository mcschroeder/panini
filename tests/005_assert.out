tests/000_assert.pan:5:1-3: error:
  • Unable to validate verification condition for f1
tests/000_assert.pan:11:1-3: error:
  • Unable to validate verification condition for f3
tests/000_assert.pan:19:1-3: error:
  • Unable to validate verification condition for f6
tests/000_assert.pan:22:1-3: error:
  • Unable to validate verification condition for f7
tests/000_assert.pan:35:1-4: error:
  • Unable to validate verification condition for f12
tests/000_assert.pan:41:1-4: error:
  • Unable to validate verification condition for f14
tests/000_assert.pan:49:1-4: error:
  • Unable to validate verification condition for f17
tests/000_assert.pan:58:1-4: error:
  • Unable to validate verification condition for f20
tests/000_assert.pan:61:1-4: error:
  • Unable to validate verification condition for f21
f0 : 𝟙
f2 : 𝟙
f4 : {_:𝟙 | true}
f5 : {_:𝟙 | false}
f8 : {b:𝔹 | b = true } → 𝟙
f9 : {b:𝔹 | true} → 𝟙
f10 : {b:𝔹 | false} → 𝟙
f11 : 𝟙
f13 : 𝟙
f15 : {_:𝟙 | true}
f16 : {_:𝟙 | false}
f18 : {s:𝕊 | false} → 𝟙
f19 : {s:𝕊 | false} → 𝟙
f22 : 𝕊 → 𝟙
f23 : {s:𝕊 | true} → 𝟙
f24 : {s:𝕊 | true} → 𝟙
f25 : {s:𝕊 | false} → 𝟙
f26 : {s:𝕊 | s = "hello"} → 𝟙